android/WALT/app/src/main/java/org/chromium/latency/walt/ScreenResponseFragment.java

```diff
@@ -215,5 +220,87 @@ public void onClick(View v) {
             return;
         }
 
+        if (v.getId() == R.id.button_brightness_curve) {
+            logger.log("Starting screen brightness curve measurement");
+            startBrightnessCurve();
+            return;
+        }
+
+    }
+
+    private ClockManager.TriggerHandler brightnessTriggerHandler = new ClockManager.TriggerHandler() {
+        @Override
+        public void onReceive(ClockManager.TriggerMessage tmsg) {
+            logger.log("ERROR: Brightness curve trigger got a trigger message, " +
+                    "this should never happen."
+            );
+        }
+
+        @Override
+        public void onReceiveRaw(String s) {
+            brightnessCurveData.append(s);
+            if (s.trim().equals("end")) {
+                // Remove the delayed callbed and run it now
+                handler.removeCallbacks(finishBrightnessCurve);
+                handler.post(finishBrightnessCurve);
+            }
+        }
+    };
+
+    void startBrightnessCurve() {
+        try {
+            clockManager.syncClock();
+            clockManager.startListener();
+        } catch (IOException e) {
+            logger.log("Error starting test: " + e.getMessage());
+            return;
+        }
+
+        clockManager.setTriggerHandler(brightnessTriggerHandler);
+
+        mBlackBox.setText("");
+
+        long tStart = clockManager.micros();
+
+        try {
+            clockManager.command(ClockManager.CMD_BRIGHTNESS_CURVE);
+        } catch (IOException e) {
+            logger.log("Error sending command CMD_BRIGHTNESS_CURVE: " + e.getMessage());
+            return;
+        }
+
+        mBlackBox.setBackgroundColor(Color.WHITE);
+
+        logger.log("=== Screen brightness curve: ===\nt_start: " + tStart);
+
+        handler.postDelayed(finishBrightnessCurve, 1000);
```

Comment:
I would prefer this as a named constant. That way it will be easier to convert to a preference.